Scarb.toml

```diff
@@ -16,6 +16,7 @@ starknet = ">=2.6.3"
 openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", tag="v0.11.0" }
 components = { path = "listings/applications/components" }
 snforge_std = { git = "https://github.com/foundry-rs/starknet-foundry.git", tag = "v0.24.0" }
+alexandria_storage = { git = "https://github.com/keep-starknet-strange/alexandria.git", rev="800f5ad" }
```

Comment:
Two reasons:
1. I moved this line from ["using lists" listing](https://github.com/NethermindEth/StarknetByExample/pull/219/files#diff-1e644d43848cb3ed6cdbcbdbafa851e88e74e31b5b2ef6b2102c7569f2cd87d0L8) and didn't want to break anything in there by changing the dependency version.
2. The most recent official Alexandria release [supports Cairo v2.6.0](https://github.com/keep-starknet-strange/alexandria/releases) and we use v2.6.3.

Wanted to test the behavior with removing this `rev` param, and it seems to _not_ break anything anywhere.
Will use the latest official release then.

Reply:
No worry with using a rev, but we should try to add a comment that explain why (not from you in this case).

> Wanted to test the behavior with removing this rev param, and it seems to not break anything anywhere.

In this case, can you still explicitly set the `tag` to the latest version to avoid unexpected resolving of future releases

Reply:
In a map, each insert and access will compute the hash of the key to get the address in memory, adding a slight total steps overhead. Also, you'll store each ContractAddress keys in storage + the index. Using List/Array also adds some overhead in terms of steps, but you still need to store all keys in storage (which make sense if your goal is to construct an array of tuples (key, value)).

I'm curious to see the result of the benchmark!

Reply:
Adding the explicit `tag` breaks the contracts, as the latest Alexandria release demands exact Cairo v2.6.0...
Reverting to use `rev` and adding a comment explaining the reason why.